### ISSUE TITLE
GeneratedCodeAnalysisFlags values need adjusting

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#9048]
The enum has three issues that need to be addressed:

- `Default` should have the value zero to be consistent with BCL guidelines.
- The values should be powers of two increasing, not 1, 16 and 17.  
- `None` should be removed. 

This bug comes from our API review meeting.

**Comment by mavasani** [comment]
Are we proposing to reverse the field semantics, i.e.

```c#
public enum GeneratedCodeAnalysisFlags
{

    /// <summary>
    /// Default analysis mode for generated code.
    /// </summary>
    /// <remarks>
    /// This mode will always guarantee that analyzer action callbacks are enabled for generated code, i.e. <see cref="Analyze"/> will be set.
    /// However, the default diagnostic reporting mode is liable to change in future.
    /// </remarks>
    Default = 0x00,

    /// <summary>
    /// Disable analyzer action callbacks for generated code.
    /// </summary>
    DisablyAnalysis = 0x01,

    /// <summary>
    /// Do not report diagnostics on generated code.
    /// </summary>
    SuppressDiagnostics = 0x10, 
}
```


Note that this requirement: _Default should have the value zero to be consistent with BCL guidelines._ would mean that we will never be able to turn off suppressing diagnostics on generated code by default even in future when we implement the editorconfig story and allow users to configure what classifies as generated code for their project.

Also even if we changed the default to SuppressDiagnostics, how would the analyzer author for a generated code analyzer explicitly turn on reporting diagnostics on generated code?

Can you please provide the exact proposal and desired semantics here?

**Comment by jaredpar** [comment]
CC @srivatsn

**Comment by mavasani** [comment]
Fixed with #9137
